lib/plugins.py

```diff
@@ -564,6 +600,12 @@ def close(self):
             if not l:
                 hooks.pop(name, None)
         self._hooks_i_registered.clear()  # just to kill strong refs to self ASAP, for GC
+        # remove registered daemon commands
+        for cmdname in self._daemon_commands:
+            try:
+                del self.parent.daemon_commands[k]
+            except KeyError:
+                pass # again, shouldn't happen
```

Comment:
Why not also do `self._daemon_command.clear()` here when done?  (Again, for same reason it was done above -- GC purposes, etc).

Reply:
Ah this is just a list of strings so it's no issue.

Reply:
Oh, lol. ok :) Still tho.. what about defensive programming? Like kill the thing because who knows what evil lurks in other parts of the code where plugins can get "closed" twice, etc (that would be a bug in the plugin subsystem obviously but who knows..?)

Reply:
Ok, let me try again since I fear this will get ignored: please please pretty please just empty that tuple when done with it?  A byte saved is a byte earned.  And also "mah invariants".  Etc.. Please?

Reply:
Oh sorry, I missed this comment...
I still think it's fine... we don't delete most instance variables, just the ones that would cause a cycle.

Reply:
true. I'm just a control freak.